### PR TITLE
Expose a rake task to `npm install`

### PR DESCRIPTION
**Pull request by seanpdoyle**
Assumes either:
* `bower` is a globally accessible command
* `package.json` includes:
```json
"tasks": {
  "postinstall": "/path/to/bower/bin/bower install"
}
```

Corresponds to https://github.com/rwz/ember-cli-rails/issues/33

### CODE REVIEW

**Comment by rondale-sc** [comment]
@seanpdoyle This looks really promising.  I'm going to test this today or tomorrow and talk to @rwz about it. 

/cc @rwz

**Comment by seanpdoyle** [comment]
Great! I really dislike the fact that you need to depend on `postinstall` pointing to a bower, but its a start

File: lib/ember-cli/helpers.rb [hunk 1]
```diff
@@ -25,10 +25,18 @@ def non_production?
       !Rails.env.production? && Rails.configuration.consider_all_requests_local
     end
 
+    def inject_install_dependencies_task!
```
Comment (rwz) [inline lib/ember-cli/helpers.rb]: Could we move the definition if this task into a separate file? Kind if like the way it's described [here](http://blog.nathanhumbert.com/2010/02/rails-3-loading-rake-tasks-from-gem.html)

File: lib/ember-cli/app.rb [hunk 2]
```diff
@@ -16,6 +16,10 @@ def compile
       silence_stream(STDOUT){ exec command }
     end
 
+    def install_dependencies
+      exec "npm install"
```
Comment (rwz) [inline lib/ember-cli/app.rb]: How about we use our `which` helper to locate the `npm` bin here? We could cache it in configuration as well.

Something like this:

```ruby
class EmberCLI::Configuration
  def npm_path
    @npm_path ||= which("npm")
  end
end

class EmberCLI::App
  delegate :npm_path, to: :configuration
  def install_dependencies
    exec "#{npm_path} install"
  end
end

**Comment by rwz** [comment]
Looks good to me!

**Comment by rondale-sc** [comment]
:+1:

**Comment by rwz** [comment]
Yay! Now I guess we need to describe this process in README somehow.

**Comment by rwz** [comment]
Thanks forPR @seanpdoyle

**Comment by rondale-sc** [comment]
@seanpdoyle Yeah, thanks!  :beers:

**Pull request merged**